gnucash/gnome/gnc-plugin-page-register.c

```diff
@@ -1264,11 +1275,15 @@ gnc_plugin_page_register_create_widget (GncPluginPage *plugin_page)
     }
 
     /* Update Query with Filter Status and Dates */
-    gnc_ppr_update_status_query (page, FALSE);
-    gnc_ppr_update_date_query(page, FALSE);
+    gnc_ppr_update_status_query (page);
+    gnc_ppr_update_date_query (page);
 
-    /* Now do the refresh */
-    gnc_ledger_display_refresh(priv->ledger);
+    /* Now do the refresh if this is a new page instaed of restore */
+    if (create_new_page)
+    {
+        priv->enable_refresh = TRUE;
+        gnc_ledger_display_refresh(priv->ledger);
+    }
 
```

Comment:
Instead, move the ```if (priv->enable_refresh)``` block from above down here:
```C
/* We always want to call gnc_ppr_update_status_query and gnc_ppr_update_date_query with refresh disabled here */
     if (priv->enable_refresh)
     {
           create_new_page = TRUE;
           priv->enable_refresh = FALSE;
      }
      gnc_ppr_update_status_query (page);
      gnc_ppr_update_date_query (page);
      if (create_new_page)
      {
           priv->enable_refresh = TRUE;
           gnc_ledger_display_refresh(priv->ledger);
      }
```

That keeps the machination all in one place so it's clearer what's going on.

Reply:
However, ISTM it would be be a better design to not have gnc_ledger_display_refresh() as a side effect of gnc_ppr_update_status_query, gnc_ppr_update_date_query, gnc_split_reg_change_style, and gnc_split_reg_double_line and to instead call it explicitly when it's the right thing to do and not otherwise.

Reply:
Moved as suggested.